src/plugins/vis_type_vega/public/vega_view/vega_map_view.js

```diff
@@ -35,30 +59,15 @@ export class VegaMapView extends VegaBaseView {
     let limitMinZ = 0;
     let limitMaxZ = 25;
 
+    // In some cases, Vega may be initialized twice, e.g. after awaiting...
+    if (!this._$container) return;
+
     if (mapConfig.mapStyle !== false) {
-      const tmsServices = await this._serviceSettings.getTMSServices();
-      // In some cases, Vega may be initialized twice, e.g. after awaiting...
-      if (!this._$container) return;
-      const emsTileLayerId = getEmsTileLayerId();
-      const mapStyle =
-        mapConfig.mapStyle === 'default' ? emsTileLayerId.bright : mapConfig.mapStyle;
       const isDarkMode = getUISettings().get('theme:darkMode');
-      baseMapOpts = tmsServices.find((s) => s.id === mapStyle);
-      baseMapOpts = {
-        ...baseMapOpts,
-        ...(await this._serviceSettings.getAttributesForTMSLayer(baseMapOpts, true, isDarkMode)),
-      };
-      if (!baseMapOpts) {
-        this.onWarn(
-          i18n.translate('visTypeVega.mapView.mapStyleNotFoundWarningMessage', {
-            defaultMessage: '{mapStyleParam} was not found',
-            values: { mapStyleParam: `"mapStyle": ${JSON.stringify(mapStyle)}` },
-          })
-        );
-      } else {
```

Comment:
impossible case. for all cases `if (!baseMapOpts) {` returns false

Reply:
Agreed that this logic is an impossible case (which is causing a bug), but a warning should be logged if `const mapOptions = tmsServices.find((s) => s.id === mapStyle);` returns undefined.

To view the issue, add `map.includeElasticMapsService: false` to your kibana.dev.yml. Then remove `map.tilemap.url` as well. This will disable EMS so mapOptions will be undefined. Then, running vega will fail because baseMapOpts is not properly defined
<img width="400" alt="Screen Shot 2021-01-14 at 7 12 07 AM" src="https://user-images.githubusercontent.com/373691/104602158-56d6b100-5638-11eb-9d3d-b5336416afd3.png">

I would recommend returning null from `getMapStyleOptions` if `mapOptions` is undefined. Then, add logic to the caller of `getMapStyleOptions` to check for null and log the warning if the return is null.

I am ok if you do not want to fix this issue in this PR, instead, we can just create a new issue and tackle this is separate thread.

Reply:
was fixed it that PR